src/lib/utils/JsonEncoder.ts

```diff
@@ -1,11 +1,67 @@
+import { base64ToBase64URL } from './base64';
+import { Buffer } from './buffer';
+
 export class JsonEncoder {
-  // TODO I would rather see here use of `object` instead of `unknown` but that's banned by @typescript-eslint/ban-types.
-  // Type `Record<string, unknown>` returns weird error "Index signature is missing in type 'Cred'.".
-  public static toBase64(data: unknown) {
-    return Buffer.from(JSON.stringify(data), 'utf-8').toString('base64');
+  /**
+   * Encode json object into base64 string.
+   *
+   * @param json the json object to encode into base64 string
+   */
+  public static toBase64(json: unknown) {
+    return JsonEncoder.toBuffer(json).toString('base64');
+  }
+
+  /**
+   * Encode json object into base64url string.
+   *
+   * @param json the json object to encode into base64url string
+   */
+  public static toBase64URL(json: unknown) {
+    return base64ToBase64URL(JsonEncoder.toBase64(json));
   }
 
+  /**
+   * Decode base64 string into json object. Also supports base64url
+   *
+   * @param base64 the base64 or base64url string to decode into json
+   */
   public static fromBase64(base64: string) {
-    return JSON.parse(Buffer.from(base64, 'base64').toString('utf-8'));
+    return JsonEncoder.fromBuffer(Buffer.from(base64, 'base64'));
+  }
+
+  /**
+   * Encode json object into string
+   *
+   * @param json the json object to encode into string
+   */
+  public static toString(json: unknown) {
+    return JSON.stringify(json);
+  }
+
+  /**
+   * Decode string into json object
+   *
+   * @param string the string to decode into json
+   */
+  public static fromString(string: string) {
+    return JSON.parse(string);
+  }
```

Comment:
Maybe. We're already using the JSON Encoder for all the other stuff so I thought it will be consistent in a way that everything that has to do with JSON encoding is done by the `JsonEncoder`.

I'm fine with removing it